### PR TITLE
Add destroy method for jQueryAutoCompleteControl to clean listeners

### PR DESCRIPTION
**Pull request by michalvankodev**
When using components which are added/removed through the use of the application which use this autocomplete-box controller they leak memory caused by not cleaned listener and not destroyed jQuery autocomplete widget.

### CODE REVIEW

File: brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js [hunk 1]
```diff
@@ -11,8 +11,6 @@ var Core = require('br/Core');
  * @module br/presenter/control/selectionfield/JQueryAutoCompleteControl
  */
 
-var jQuery = require('jquery');
```
Comment (janhancic) [inline brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js]: Why did you remove this? It's a dependency.

Comment (michalvankodev) [inline brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js]: Its duplicated .. jQuery is defined as first dependency at the top

File: brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js [hunk 2]
```diff
@@ -163,4 +161,12 @@ JQueryAutoCompleteControl.prototype.setOptions = function(mOptions) {
 	}
 };
 
+/**
+ * Destroy created listeners and jQuery autocomplete plugin
+ */
+JQueryAutoCompleteControl.prototype.destroy = function() {
+	this.m_oPresentationNode.value.removeListener(this._valueChangedListener);
```
Comment (thecapdan) [inline brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js]: @michalvankodev can we add a gueard here in case detroy is called without any nodes being set... e.g. if(this._valueChangedListener). similarly a guard should be added for this.m_eElement on the line below

Comment (michalvankodev) [inline brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js]: Yes, Today we found out that not all Nodes that use jQueryAutoCompleteControl had triggered `onViewReady` and then it will break the execution because the `autocomplete` plugin will throw error as no autocomplete was set.

Comment (michalvankodev) [inline brjs-sdk/sdk/libs/javascript/br-presenter/src/br/presenter/control/selectionfield/JQueryAutoCompleteControl.js]: I've updated the PR wth the checks.
